strongbox-web-core/src/test/java/org/carlspring/strongbox/rest/ArtifactControllerTest.java

```diff
@@ -0,0 +1,761 @@
+    {
+        String ga = "org.carlspring.strongbox.metadata:metadata-foo";
+
+        Artifact artifact1 = ArtifactUtils.getArtifactFromGAVTC(ga + ":3.1-SNAPSHOT");
+        TestCaseWithArtifactGeneration generator = new TestCaseWithArtifactGeneration();
+        Artifact artifact1WithTimestamp1 = ArtifactUtils.getArtifactFromGAVTC(
+                ga + ":" + generator.createSnapshotVersion("3.1", 1));
+        Artifact artifact1WithTimestamp2 = ArtifactUtils.getArtifactFromGAVTC(
+                ga + ":" + generator.createSnapshotVersion("3.1", 2));
+        Artifact artifact1WithTimestamp3 = ArtifactUtils.getArtifactFromGAVTC(
+                ga + ":" + generator.createSnapshotVersion("3.1", 3));
+        Artifact artifact1WithTimestamp4 = ArtifactUtils.getArtifactFromGAVTC(
+                ga + ":" + generator.createSnapshotVersion("3.1", 4));
+
+        // artifactDeployer.setClient(client);
+
+        String storageId = "storage0";
+        String repositoryId = "snapshots";
+
+        generateAndDeployArtifact(artifact1WithTimestamp1, storageId, repositoryId);
+        generateAndDeployArtifact(artifact1WithTimestamp2, storageId, repositoryId);
+        generateAndDeployArtifact(artifact1WithTimestamp3, storageId, repositoryId);
+        generateAndDeployArtifact(artifact1WithTimestamp4, storageId, repositoryId);
+
+    /*    Metadata versionLevelMetadata = client.retrieveMetadata("storages/" + storageId + "/" + repositoryId + "/" +
+                                                                ArtifactUtils.getVersionLevelMetadataPath(artifact1));*/
+        String path = "storages/" + storageId + "/" + repositoryId + "/" +
+                      ArtifactUtils.getVersionLevelMetadataPath(artifact1);
+
+
+        InputStream is = getArtifactAsStream(ArtifactUtils.getVersionLevelMetadataPath(artifact1),
+                                             "storages/" + storageId + "/" + repositoryId);
+        MetadataXpp3Reader reader = new MetadataXpp3Reader();
+        Metadata versionLevelMetadata = reader.read(is);
+
+        Assert.assertNotNull(versionLevelMetadata);
+        Assert.assertEquals("org.carlspring.strongbox.metadata", versionLevelMetadata.getGroupId());
+        Assert.assertEquals("metadata-foo", versionLevelMetadata.getArtifactId());
+        Assert.assertEquals(4, versionLevelMetadata.getVersioning().getSnapshot().getBuildNumber());
+        Assert.assertNotNull(versionLevelMetadata.getVersioning().getLastUpdated());
+        Assert.assertEquals(12, versionLevelMetadata.getVersioning().getSnapshotVersions().size());
+    }
+
+    /**
+     * Looks up a storage by it's ID.
+     */
+    private void generateAndDeployArtifact(Artifact artifact,
+                                           String storageId,
+                                           String repositoryId,
+                                           ArtifactDeployer artifactDeployer)
+            throws FileNotFoundException, NoSuchAlgorithmException {
```

Comment:
Please, move the braces to the next line for all the methods like this one below.